test/clean.js

```diff
@@ -31,22 +40,27 @@ describe( 'clean', function () {
 
 	it( '(preparing: remove files to clean)', function ( done ) {
 
-		rmdir( 'test/fixtures/cleaning', done );
+		rmdir( 'test/fixtures/cleanthis', done );
+
+	} );
+
+	it( '(preparing: remove files to ignore in cleanup)', function ( done ) {
+
+		rmdir( 'test/fixtures/dontcleanthis', done );
 
 	} );
 
-	it( 'should clean up extra remote files', function ( done ) {
+	it( 'should clean up extra remote files but repect ignore rules', function ( done ) {
 
 		done = suite.done( done );
 
-		suite.vftp.clean( 'test/clean/**', 'test/fixtures' )
+		suite.vftp.clean( [ 'test/clean/**/*', '!test/clean/dontcleanthis/**/*' ], 'test/fixtures' )
```

Comment:
This operation doesn't quite work yet, and the problem may lie with the implementation of the `clean` function. Skimming its implementation, it appears a directory can be deleted even though one or more of its containing files is excluded from the clean operation. I don't have enough insight into the code to fix this quickly, so could use some help here.